src/switch_core_media.c

```diff
@@ -1939,16 +1962,20 @@ static switch_rtp_crypto_mode_t switch_core_session_check_crypto_prefs(switch_co
 	const char *var = NULL;
 	const char *val = NULL;
 	char *suites = NULL;
+	switch_media_handle_t *smh = session->media_handle;
 
 	if (switch_channel_direction(session->channel) == SWITCH_CALL_DIRECTION_INBOUND) {
 		var = "rtp_secure_media_inbound";
 	} else {
 		var = "rtp_secure_media_outbound";
 	}
 
-	if (!(val = switch_channel_get_variable(session->channel, var))) {
+	if (!(val = switch_channel_get_variable(session->channel, var)) || (!smh && !zstr(smh->mparams->rtp_secure_media))) {
```

Comment:
The validation for rtp_secure_media for smh should be done in the if statement below.